src/app/pages/question/question.component.ts

```diff
@@ -47,21 +48,40 @@ export class QuestionComponent implements OnInit, CanDeactivate<CanComponentDeac
 
   canDeactivate(): Observable<boolean> {
     if (this.currentIndex$.value == QUESTIONS_SIZE) {
-      this.modalWindowData$ = new BehaviorSubject<ModalWindowModel>(this.modalService.getData(ModalRoutes.Finish));
+      this.nextRoute = '/finish';
     }
-    const item = event?.target as HTMLElement;
-    this.modalWindowData$ = new BehaviorSubject<ModalWindowModel>(this.modalService.getData(item.textContent!));
+    
+    this.modalWindowData$ = new BehaviorSubject<ModalWindowModel>(this.modalService.getData(this.nextRoute));
     this.modalWindowState$.next(true);
     if (this.modalWindowState$.value && this.modalWindowChoice$.value) {
       return of(true);
     }
     return of(false);
   }
 
+  @HostListener('window:beforeunload', ['$event'])
+  unloadNotification($event: BeforeUnloadEvent): void {
+    $event.preventDefault();
+  }
+
+  @HostListener('document:keydown', ['$event'])
```

Comment:
I believe we can omit this, because `beforeunload` event works also for keyboard shortcuts